token.js

```diff
@@ -1,33 +1,54 @@
-const ADDRESS = "0xff56cc6b1e6ded347aa0b7676c85ab0b3d08b0fa";
-const ABI = require("./abi.json");
+const ERC20_CONTRACT_ADDRESS = "0xff56cc6b1e6ded347aa0b7676c85ab0b3d08b0fa";
+const STAKING_ADDRESS = "0x01D59Af68E2dcb44e04C50e05F62E7043F2656C3";
+
+const ERC20_ABI = require("./token.abi.json");
+const STAKING_ABI = require("./staking.abi.json");
 
 const LONG_TERM_RESERVES = "0x384f5cf955b39b76c47a440f14c31ad39fd39d00";
 const PRIVATE_SALE = "0x1bef7f8798256e033eaa42f005d2b53079b90ffe";
 const TEAM_AND_FOUNDING_PARTNERS = "0xc200f98f3c088b868d80d8eb0aeb9d7ee18d604b";
 const ADVISORS = "0x574d48b2ec0a5e968adb77636656672327402634";
 
-function getContract(web3) {
-    return new web3.eth.Contract(ABI, ADDRESS);
+const NON_CIRCULATING_WALLETS = [LONG_TERM_RESERVES, PRIVATE_SALE, TEAM_AND_FOUNDING_PARTNERS, ADVISORS];
+
+function getERC20Contract(web3) {
+    return new web3.eth.Contract(ERC20_ABI, ERC20_CONTRACT_ADDRESS);
+}
+
+function getStakingContract(web3) {
+    return new web3.eth.Contract(STAKING_ABI, STAKING_ADDRESS);
+}
+
+function sum(list) {
+    return list.reduce((previousValue, currentValue) => {
+        return previousValue + currentValue;
+    }, BigInt(0));
 }
 
 async function tokensInCirculation(web3) {
-    const contract = getContract(web3);
+    const erc20 = getERC20Contract(web3);
+    const staking = getStakingContract(web3);
+
     const block = await web3.eth.getBlockNumber();
     const updatedAt = new Date();
```

Comment:
If we publish update at it's preferred to use the block timestamp

Reply:
You may use:
const block = web3.eth.getBlock("latest");
const blockNumber = block.number;
const blockTime = block timestamp;

   * baseTime <- block.timestamp

Reply:
Thank you! Fixed.